### DISCUSSION TITLE
Is `brew install composer` an official and recommended way of installing composer on a Mac?

### DISCUSSION THREAD

**Discussion by nbro** [composer/composer#11970] [General]
I like to use `brew` to install software on a Mac. I don't like to user scripts or multiple commands because I will forget this stuff, while, if I use `brew`, can always check later what I installed. 

There's a `composer` formula here: https://formulae.brew.sh/formula/composer. However, it is not mentioned in [the documentation](https://formulae.brew.sh/formula/composer), which suggests to use several commands or a script to download the composer installer.

My question is in the title - is this brew formula official and so maintained by you? If yes, why don't you mention this option in the documentation? If not, why not?

Here's the documentation I read.

- https://getcomposer.org/doc/faqs/how-to-install-composer-programmatically.md
- https://getcomposer.org/download/
- https://getcomposer.org/doc/00-intro.md#downloading-the-composer-executable

**Comment by Seldaek** [comment]
It is not recommended because we previously had issues with it, and it's not under our control, so it's kind of a pain to fix. Most of the time it works fine though, so if you're used to brew it's probably fine to use it.

**Reply by nbro** [reply]
Could you point to the issues that occurred with the brew installation, if this is documented?

**Reply by Seldaek** [reply]
Sorry no it's been ages, I don't recall exactly, you might find stuff with search but it wasn't anything unfixable, and I'm sure it's fine to use brew. We just don't want to recommend it because we do not control it.
